### PR TITLE
OSDOCS-4206 Document installing OpenShift using Azure Marketplace [4.9]

### PR DESCRIPTION
**Pull request by bscott-rh**
https://issues.redhat.com/browse/OSDOCS-4206
4.9 backport of https://issues.redhat.com/browse/OSDOCS-3029

Doc preview:
[Configuring an Azure account](https://bscott-rh.github.io/openshift-docs/OSDOCS-4206/installing/installing_azure/installing-azure-account.html#installation-azure-marketplace_installing-azure-account)
[Installing on Azure with customizations](https://bscott-rh.github.io/openshift-docs/OSDOCS-4206/installing/installing_azure/installing-azure-customizations.html#installation-azure-marketplace-subscribe_installing-azure-customizations)
[Installing on Azure with customizations - updating manifests](https://bscott-rh.github.io/openshift-docs/OSDOCS-4206/installing/installing_azure/installing-azure-customizations.html#installation-azure-marketplace-manifests_installing-azure-customizations)
[Installing with UPI](https://bscott-rh.github.io/openshift-docs/OSDOCS-4206/installing/installing_azure/installing-azure-user-infra.html#installation-azure-marketplace-subscribe_installing-azure-user-infra)

### CODE REVIEW

**Comment by bscott-rh** [comment]
@MayXuQQ PTAL at this 4.9 backport for Azure Marketplace installation doc. Thank you

File: modules/installation-azure-marketplace-subscribe.adoc [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+----
+** EMEA:
++
+[source,terminal]
+----
+$ az vm image terms accept --urn redhat-limited:rh-ocp-worker:rh-ocp-worker:<version>
+----
+
+ifdef::ipi[]
+. Record the image details of your offer. You must update the `99_openshift-cluster-api_worker-machineset-[0-2].yaml` files in the section titled "Updating Manifests for Marketplace Installation" before completing the installation.
```
Comment (MayXuQQ) [inline modules/installation-azure-marketplace-subscribe.adoc:107]: need update 06_workers_market_place.json, not `99_openshift-cluster-api_worker-machineset-[0-2].yaml`

Comment (bscott-rh) [reply modules/installation-azure-marketplace-subscribe.adoc:107]: According to the steps in [this KCS](https://access.redhat.com/articles/6484031), IPI users should modify the installation manifests and UPI users should modify the ARM templates. Is that KCS incorrect?

Comment (MayXuQQ) [reply modules/installation-azure-marketplace-subscribe.adoc:107]: sorry, for UPI is 06_workers_market_place.json, for IPI is 99_openshift-cluster-api_worker-machineset-[0-2].yaml

**Comment by bscott-rh** [comment]
/label peer-review-needed

This is one of three PR's backporting the same content to 4.8, 4.9 and 4.10. Whoever reviews this may want to pick up all 3 as the content is essentially identical. 

4.8: https://github.com/openshift/openshift-docs/pull/50828
4.9: https://github.com/openshift/openshift-docs/pull/50895
4.10: https://github.com/openshift/openshift-docs/pull/51370

**Review by tmalove — COMMENTED** [review]
/remove-label peer-review-needed
/label peer-review-in-progress

File: modules/installation-azure-marketplace-manifests.adoc [hunk 2]
```diff
@@ -0,0 +1,45 @@
+// Module included in the following assemblies:
+//
+// * installing/installing_azure/installing-azure-customizations.adoc
+
+:_content-type: PROCEDURE
+[id="installation-azure-marketplace-manifests_{context}"]
+= Updating Manifests for Marketplace Installation
```
Comment (tmalove) [inline modules/installation-azure-marketplace-manifests.adoc:7]:
```suggestion
= Updating Manifests for Marketplace installation
```

File: modules/installation-azure-marketplace.adoc [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+[id="installation-azure-marketplace_{context}"]
+= Supported Azure Marketplace regions
+
+Installing a cluster using the Azure Marketplace image is available to customers who purchase the offer in North America and EMEA.
+
+While the offer must be purchased in North America or EMEA, you can deploy the cluster to any of the Azure public partitions that {product-title} supports.
+
+[NOTE]
+====
+Deploying a cluster using the Azure Marketplace image is not supported for the Azure Government regions.
```
Comment (tmalove) [inline modules/installation-azure-marketplace.adoc:15]:
```suggestion
Deploying a cluster by using the Azure Marketplace image is not supported for the Azure Government regions.
```

File: modules/installation-azure-marketplace-subscribe.adoc [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+endif::ipi[]
+
+ifdef::upi[]
+. Record the image details of your offer and use them to update the `06_workers.json` Azure Resource Manager (ARM) template. 
+
+. Update the `storageProfile.imageReference` field by deleting the `id` parameter and adding the `offer`, `publisher`, `sku`, and `version` parameters using the values from your offer. You can find a sample template in the "Creating additional worker machines in Azure" section.
```
Comment (tmalove) [inline modules/installation-azure-marketplace-subscribe.adoc:113]:
```suggestion
. Update the `storageProfile.imageReference` field by deleting the `id` parameter and adding the `offer`, `publisher`, `sku`, and `version` parameters by using the values from your offer. You can find a sample template in the "Creating additional worker machines in Azure" section.
```

**Review by tmalove — COMMENTED** [review]
/remove-label peer-review-in-progress
/label peer-review-done

**Pull request merged**